src/components/Searcher.js

```diff
@@ -970,6 +989,44 @@ class Searcher extends React.Component<SearcherDefaultProps, SearcherProps, Sear
     });
   }
 
+  /**
+   * Add signal each time a filter is added/removed.
+   * When a filter is added, signal with weight 1 is created.
+   * When a filter is removed, signal with weight 0 is created.
+   */
+  addFacetFilterSignal(facetFilter: FacetFilter, weight: number) {
+    const queryDocuments = this.state.response ? this.state.response.documents : null;
+    const querySignal = queryDocuments ? queryDocuments[0].signal : null;
+    const facets = this.state.response ? this.state.response.facets : null;
+    if (!querySignal || !facets) {
+      return;
+    }
+    const facet = facets.find((searchFacet) => {
+      return searchFacet.label === facetFilter.facetName;
+    });
+    if (!facet) {
+      return;
+    }
+    const signal = new SignalData();
+    signal.docId = facetFilter.filter;
+    signal.docOrdinal = facet.buckets.findIndex((bucket) => {
+      return bucket.filter === facetFilter.filter;
+    }) + 1; // index starts at 1
+    signal.featureVector = '';
+    signal.locale = querySignal.locale;
+    signal.principal = querySignal.principal;
+    signal.query = querySignal.query;
+    signal.queryTimestamp = querySignal.queryTimestamp;
+    signal.relevancyModelName = querySignal.relevancyModelName;
+    signal.relevancyModelNames = querySignal.relevancyModelNames;
+    signal.relevancyModelVersion = querySignal.relevancyModelVersion;
+    signal.signalTimestamp = Date.now();
+    signal.ttl = false;
+
+    const doc = new SearchDocument(new Map(), signal);
+    new Signals(this.props.baseUri).addSignal(doc, 'facet', weight);
```

Comment:
We should maybe refactor the `addSignal` method so it takes a signal instead of a document; it seems silly to have to create a document that the method doesn't really care about.
If we do this, we should make sure it is documented as a breaking change, but I think that's fine considering that the React 16 change is in here too... (I doubt anybody is calling this from their application code but better safe than sorry...)